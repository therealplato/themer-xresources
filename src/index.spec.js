import { render } from './index';
import { colors } from 'themer-colors-default';
import path from 'path';

describe('Hyper.app theme generator', () => {

  const promisedFiles = Promise.all(render(colors, {}));

  it('should render four files when given a color set with both a light and a dark theme', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(4);
    expect(files.filter(file => path.basename(file.name) === 'package.json').length).toBe(2);
    expect(files.filter(file => path.basename(file.name) === 'index.js').length).toBe(2);
  });

  it('should render valid JSON to package.json files', async () => {
    const files = await promisedFiles;
    files.filter(file => path.basename(file.name) === 'package.json').map(
      file => JSON.parse(file.contents.toString('utf8'))
    ).forEach(packageJson => {
      delete packageJson.version;
      expect(packageJson).toMatchSnapshot();
    });
  });

  it('should render valid plugin files', async () => {
    const files = await promisedFiles;
    files.filter(file => path.basename(file.name) !== 'package.json').forEach(
      file => expect(file.contents.toString('utf8')).toMatchSnapshot()
    );
  });

  it('should provide promised files whose contents are buffers', async () => {
    const files = await promisedFiles;
    files.forEach(file => expect(file.contents).toBeInstanceOf(Buffer));
  });

});
