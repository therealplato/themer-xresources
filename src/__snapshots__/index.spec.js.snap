// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hyper.app theme generator should render valid JSON to package.json files 1`] = `
Object {
  "description": "dark theme for Hyper.app, generated by themer",
  "keywords": Array [
    "themer",
    "dark",
    "hyper",
  ],
  "main": "themer-hyper-dark/index.js",
  "name": "themer-hyper-dark",
}
`;

exports[`Hyper.app theme generator should render valid JSON to package.json files 2`] = `
Object {
  "description": "light theme for Hyper.app, generated by themer",
  "keywords": Array [
    "themer",
    "light",
    "hyper",
  ],
  "main": "themer-hyper-light/index.js",
  "name": "themer-hyper-light",
}
`;

exports[`Hyper.app theme generator should render valid plugin files 1`] = `
"
    module.exports.decorateConfig = config => {
      return Object.assign({}, config, {
        cursorColor: 'rgba(204, 120, 250, 0.5)',
        foregroundColor: '#E0DCE0',
        backgroundColor: '#282629',
        borderColor: '#282629',
        colors: {
          black: '#282629',
          red: '#FF4050',
          green: '#A4CC35',
          yellow: '#FFD24A',
          blue: '#66BFFF',
          magenta: '#F553BF',
          cyan: '#26C99E',
          white: '#E0DCE0',
          lightBlack: '#474247',
          lightRed: '#F28144',
          lightGreen: '#A4CC35',
          lightYellow: '#FFD24A',
          lightBlue: '#66BFFF',
          lightMagenta: '#F553BF',
          lightCyan: '#26C99E',
          lightWhite: '#FFFCFF',
        },
      });
    };
  "
`;

exports[`Hyper.app theme generator should render valid plugin files 2`] = `
"
    module.exports.decorateConfig = config => {
      return Object.assign({}, config, {
        cursorColor: 'rgba(191, 101, 240, 0.5)',
        foregroundColor: '#474247',
        backgroundColor: '#FFFCFF',
        borderColor: '#FFFCFF',
        colors: {
          black: '#FFFCFF',
          red: '#F03E4D',
          green: '#97BD2D',
          yellow: '#EEBA21',
          blue: '#53A6E1',
          magenta: '#EE4EB8',
          cyan: '#1FC598',
          white: '#474247',
          lightBlack: '#E0DCE0',
          lightRed: '#F37735',
          lightGreen: '#97BD2D',
          lightYellow: '#EEBA21',
          lightBlue: '#53A6E1',
          lightMagenta: '#EE4EB8',
          lightCyan: '#1FC598',
          lightWhite: '#282629',
        },
      });
    };
  "
`;
